tests/specialization_constants/specialization_constants_multiple.h

```diff
@@ -0,0 +1,215 @@
+                                                                {kernelId});
+            if (!k_bundle.has_kernel(kernelId)) {
+            log.note("kernel_bundle doesn't contain target kernel;"
+                    "multiple spec const for " +
+                    type_name_string<T>::get(type_name) + "(skipped)");
+            return;
+          }
+
+          k_bundle.template set_specialization_constant<
+              sc_multiple<T, def_values[0]>>(ref1);
+          k_bundle.template set_specialization_constant<
+              sc_multiple<T, def_values[1]>>(ref2);
+          k_bundle.template set_specialization_constant<
+              sc_multiple<T, def_values[2]>>(ref3);
+
+          auto exec_bundle = sycl::build(k_bundle);
+          cgh.use_kernel_bundle(exec_bundle);
+
+          cgh.single_task<sc_kernel_multiple<T, via_kb>>(
+              [=](sycl::kernel_handler h) {
+                get_values_from_handler<T>(res_acc, h);
+              });
+        } else {
+          // No kernel_bundle
+          cgh.set_specialization_constant<sc_multiple<T, def_values[0]>>(ref1);
+          cgh.set_specialization_constant<sc_multiple<T, def_values[1]>>(ref2);
+          cgh.set_specialization_constant<sc_multiple<T, def_values[2]>>(ref3);
+
+          cgh.single_task<sc_kernel_multiple<T, via_kb>>(
+              [=](sycl::kernel_handler h) {
+                get_values_from_handler<T>(res_acc, h);
+              });
+        }
+      });
+    }
+    if (!check_equal_values(ref1, result_vec[0]) ||
+        !check_equal_values(ref2, result_vec[1]) ||
+        !check_equal_values(ref3, result_vec[2]) ||
+        !check_equal_values(T(value_helper<T>(def_values[3])), result_vec[3]) ||
+        !check_equal_values(T(value_helper<T>(def_values[4])), result_vec[4]) ||
+        !check_equal_values(T(value_helper<T>(def_values[5])), result_vec[5]))
+      FAIL(log,
+           "multiple spec const for " + type_name_string<T>::get(type_name));
+
+    free(result_vec);
+  }
+};
+
+template <typename via_kb>
+static void sc_run_test_core(util::logger &log) {
+  using namespace specialization_constants_multiple;
+  try {
+#ifndef SYCL_CTS_FULL_CONFORMANCE
+    for_all_types<check_specialization_constants_multiple_for_type, via_kb>(
+        get_spec_const::testing_types::types, log);
+#else
+    for_all_types_vectors_marray<
+        check_specialization_constants_multiple_for_type, via_kb>(
+        get_spec_const::testing_types::types, log);
+#endif
+    for_all_types<check_specialization_constants_multiple_for_type, via_kb>(
+        get_spec_const::testing_types::composite_types, log);
+
+  } catch (const sycl::exception &e) {
+    log_exception(log, e);
+    std::string errorMsg =
+        "a SYCL exception was caught: " + std::string(e.what());
+    FAIL(log, errorMsg);
+  } catch (const std::exception &e) {
+    std::string errorMsg = "an exception was caught: " + std::string(e.what());
+    FAIL(log, errorMsg);
+  }
```

Comment:
Move this in a generic test executor taking a lambda to execute?

Reply:
Ronan, i suggest move this try-catch block to `util/test_base.h` and make change across all project, and due to this block duplicated across all project i suggest to move this changes into separate PR
what do you think about this?

Reply:
Perfect!